geps/gep-3779/index.md

```diff
@@ -42,20 +37,669 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+//     at connection time (e.g., source identity, port). Recommended for broad,
+//     coarse-grained access controls.
+//
+//     NOTE FOR REVIEWRS -- THIS IS FOR FUTURE ENHANCEMENT
+//
+//   - Application: Enforces the policy at the application layer (L7), typically at
+//     HTTP/gRPC-aware sidecars or L7 proxies. Enables fine-grained authorization
+//     using protocol-specific attributes (e.g., HTTP paths, methods).
+//
+// This field clarifies policy intent and informs where enforcement is expected
+// to happen. It also enables implementation-specific validation and behavior.
+//
+// +kubebuilder:validation:Enum=Network;Application
+type EnforcementLevel string
+```
+
+### API Design
+
+```go
+
+type AuthorizationPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of AuthorizationPolicy.
+    Spec AuthorizationPolicySpec `json:"spec,omitempty"`
+
+    // Status defines the current state of AuthorizationPolicy.
+    Status PolicyStatus `json:"status,omitempty"`
+}
+
+// AuthorizationPolicyAction specifies the action to take.
+// +kubebuilder:validation:Enum=ALLOW;DENY
+type AuthorizationPolicyAction string
+
+const (
+    // ActionAllow allows requests that match the policy rules.
+    ActionAllow AuthorizationPolicyAction = "ALLOW"
+    
+    // NOTE FOR REVIEWERS AGAIN: GAMMA leads met on 24th July and DENY action is likely going to be pushed to 1.5 and will potentially be extended conformance. 
+
+    // ActionDeny denies requests that match the policy rules.
+    ActionDeny AuthorizationPolicyAction = "DENY"
+)
+
+// AuthorizationPolicySpec defines the desired state of AuthorizationPolicy.
+type AuthorizationPolicySpec struct {
+    // TargetRef identifies the resource this policy is attached to.
+    // (will be translated to CEL later):
+    //   - When Kind is Pod, Name MUST be Empty, Selector MUST be set, Max targetRef length is 1. 
+    //   - Otherwise multiple refs as long as Kind isn't Pod, selector MUST NOT be set
+    // +kubebuilder:validation:Required
+    TargetRefs []gatewayv1.PolicyTargetReferenceWithLabelSelectors `json:"targetRefs"`
+
+    // Action specifies the action to take when a request matches the rules.
+    // +kubebuilder:validation:Required
+    Action AuthorizationPolicyAction `json:"action"`
+
+    // Rules defines the list of authorization rules.
+    // A request matches the policy if it matches any of these rules.
+    // +optional
+    Rules []AuthorizationRule `json:"rules,omitempty"`
+}
+
+// AuthorizationRule defines a single authorization rule.
+// A request matches the rule if it matches ALL fields specified.
+type AuthorizationRule struct {
+    // AuthorizationSource specifies who is making the request.
+    // If omitted, matches any source.
+    // +optional
+    AuthorizationSource *AuthorizationSource `json:"authorizationSource,omitempty"`
```

Comment:
It's better to keep it just "Source" as it is self explanatory as it is in AuthoriztaionPolicy.